cli/src/main.rs

```diff
@@ -203,12 +204,54 @@ pub async fn serve(args: ServeArgs) -> ZResult<()> {
 
         let dev = warp::path("dev").and(warp::fs::dir(""));
 
-        dev.or(warp::fs::dir(proj.dest_dir))
-            .with(cors)
-            .with(warp::compression::gzip())
+        dev.or(warp::fs::dir(
+            proj.lock().expect("Cannot get lock").dest_dir.clone(),
+        ))
+        .with(cors)
+        .with(warp::compression::gzip())
     });
 
-    server.run(http_addr).await;
+    // server.run(http_addr).await;
+    tokio::spawn(server.run(http_addr));
+
+    if args.watch && !args.no_build {
+        let wx = watchexec::Watchexec::new(move |mut action| {
+            // Filter out event that means a new build is needed
+            if action.events.iter().any(|event| {
+                event.tags.iter().any(|tag| {
+                    matches!(tag,
+                    watchexec_events::Tag::Path {
+                        path,
+                        file_type: Some(watchexec_events::FileType::File),
+                    } if Some("typ") == path.extension().and_then(|osstr| osstr.to_str()))
+                })
+            }) {
+                proj.lock()
+                    .expect("Cannot get lock")
+                    .build()
+                    .expect("Cannot build the project");
```

Comment:
It'd be better not to panic here when the project fails to build. When editing files, failures looks very common.